### PR TITLE
Swift: Add references to appropriate READMEs

### PR DESCRIPTION
**Pull request by lickel**
Some sections are cribbed heavily from https://code.cash.app/wire-support-for-swift-part-1

### CODE REVIEW

File: wire-library/wire-runtime-swift/build.gradle.kts [hunk 1]
```diff
@@ -69,6 +69,24 @@ val generateSwiftProtos by tasks.creating(JavaExec::class) {
   }
 }
 
+val generateSampleProtos by tasks.creating(JavaExec::class) {
```
Comment (lickel) [inline wire-library/wire-runtime-swift/build.gradle.kts:72]: There is _almost certainly_ a better way to do this, but I don't know what I'm doing in Gradle

File: wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+#endif
+
+extension Dinosaur : ProtoMessage {
+    public static func protoMessageTypeURL() -> String {
+        return "type.googleapis.com/squareup.dinosaurs.Dinosaur"
+    }
+}
+
+extension Dinosaur : Proto2Codable {
+    public init(from reader: ProtoReader) throws {
```
Comment (lickel) [inline wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift:59]: For the purposes of this sample code, should we `/* snip */` out the Proto/Foundation codable implementations?

File: Package.swift [hunk 3]
```diff
@@ -23,6 +23,7 @@ let package = Package(
         ),
         // The tests depend on valid protos via gradle
         // ./gradlew -p wire-library :wire-runtime-swift:generateTestProtos
+        // ./gradlew -p wire-library :wire-runtime-swift:generateSampleProtos
```
Comment (oldergod) [inline Package.swift:26]: The comment makes it sound like the tests would not pass without having run `generateSampleProtos` but is that really the case? If not, no need for the comment I think ?

File: wire-library/wire-runtime-swift/build.gradle.kts (hunk 1 above)
Comment (oldergod) [reply wire-library/wire-runtime-swift/build.gradle.kts:72]: Why not add those files in the existing `generateTestProtos` directly?

**Review by oldergod — APPROVED** [review]

**Comment by oldergod** [comment]
@lickel I've made a big change in Wire and rebased your PR. the comments are gone but still valid, let's sync next week.

File: wire-runtime-swift/build.gradle.kts [hunk 4]
```diff
@@ -69,6 +69,24 @@ val generateSwiftProtos by tasks.creating(JavaExec::class) {
   }
 }
 
+val generateSampleProtos by tasks.creating(JavaExec::class) {
```
Comment (lickel) [inline wire-runtime-swift/build.gradle.kts:72]: There's almost certainly a better way to do this but I don't know gradle

File: Package.swift [hunk 5]
```diff
@@ -22,21 +22,22 @@ let package = Package(
...
             name: "WireTests",
             dependencies: ["Wire"],
-            path: "wire-library/wire-tests-swift/src/main/swift"
+            path: "wire-tests-swift/src/main/swift"
         ),
         .testTarget(
             name: "WireCompilerTests",
             dependencies: ["WireTests"],
-            path: "wire-library/wire-tests-swift/src/test/swift"
+            path: "wire-tests-swift/src/test/swift"
```
Comment (lickel) [inline Package.swift:40]: This rebase change implies we should have an SPM GH check

**Comment by lickel** [comment]
Might be worth linking to the 3 Dinosaur.source files instead of large code dumps?

File: Package.swift (hunk 5 above)
Comment (oldergod) [reply Package.swift:40]: What's that?

Comment (lickel) [reply Package.swift:40]: Eg we have a github action to verify Cocoapods. We probably should have something for Seift Package Manager too

File: wire-runtime-swift/build.gradle.kts (hunk 4 above)
Comment (lickel) [reply wire-runtime-swift/build.gradle.kts:72]: @oldergod 

> Why not add those files in the existing generateTestProtos directly?

I think there's value in a different subdirectory, tho it for sure should just be happening as a dependency or something. 

I am very unfamiliar with Gradle; my only experience with it was XML _years_ ago.

File: Package.swift [hunk 6]
```diff
@@ -22,21 +22,22 @@ let package = Package(
             path: "wire-runtime-swift/src/main/swift"
         ),
         // The tests depend on valid protos via gradle
-        // ./gradlew -p wire-library :wire-runtime-swift:generateTestProtos
+        // ./gradlew :wire-runtime-swift:generateTestProtos
```
Comment (lickel) [inline Package.swift:25]: @oldergod 

> The comment makes it sound like the tests would not pass without having run generateSampleProtos but is that really the case? If not, no need for the comment I think ?

True, the samples _currently_ are not a dependency, and if they were we'd need the same codegen in the Cocoapods file. 

That said, I still would prefer we codegen the dinosaur sample in the repo somewhere.

Comment (oldergod) [reply Package.swift:25]: Got it

File: wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift]: Yes, let's do that and then merge 🎸

File: wire-runtime-swift/build.gradle.kts (hunk 4 above)
Comment (lickel) [reply wire-runtime-swift/build.gradle.kts:72]: Added as a dependency

File: wire-library/wire-runtime-swift/build.gradle.kts (hunk 1 above)
Comment (lickel) [reply wire-library/wire-runtime-swift/build.gradle.kts]: Added as a dependency

File: wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift (hunk 2 above)
Comment (lickel) [reply wire-library/wire-runtime-swift/src/test/swift/sample/Dinosaur.swift]: I went a different route and used some \<details\> tags for all three languages.

**Pull request merged**